namui/src/namui/manager/web/mouse_manager.rs

```diff
@@ -124,19 +127,39 @@ impl MouseCursor {
         }
     }
 }
-fn extract_buttons(mouse_event: web_sys::MouseEvent) -> HashSet<crate::MouseButton> {
+
+fn get_pressing_buttons(mouse_event: &web_sys::MouseEvent) -> HashSet<crate::MouseButton> {
     let mouse_event_buttons = mouse_event.buttons();
-    let tuples = [
+
+    const MOUSE_BUTTONS_CONVERTING_TUPLES: [(u16, crate::MouseButton); 3] = [
         (1 << 0, crate::MouseButton::Left),
         (1 << 1, crate::MouseButton::Middle),
         (1 << 2, crate::MouseButton::Right),
     ];
 
-    HashSet::from_iter(tuples.iter().filter_map(|(bit, button)| {
-        if mouse_event_buttons & bit != 0 {
-            Some(*button)
-        } else {
-            None
-        }
-    }))
+    HashSet::from_iter(
+        MOUSE_BUTTONS_CONVERTING_TUPLES
+            .iter()
+            .filter_map(|(bit, button)| {
+                if mouse_event_buttons & bit != 0 {
+                    Some(*button)
+                } else {
+                    None
+                }
+            }),
+    )
+}
+fn get_button(mouse_event: &web_sys::MouseEvent) -> crate::MouseButton {
+    let mouse_event_button = mouse_event.button() as u16;
+
+    const MOUSE_BUTTON_CONVERTING_TUPLES: [(u16, crate::MouseButton); 3] = [
+        (0, crate::MouseButton::Left),
+        (1, crate::MouseButton::Middle),
+        (2, crate::MouseButton::Right),
```

Comment:
```suggestion
        (1 << 0, crate::MouseButton::Left),
        (1 << 1, crate::MouseButton::Middle),
        (1 << 2, crate::MouseButton::Right),
```
It looks similar to [Line 134-138](https://github.com/NamseEnt/namseent/pull/102/files#diff-e19c608c42ef6d99e2323eca7c0504cd4462291935f1423228655aeb84f01ed9R134-R138), but it's little different.

If it's a typo, Maybe you should use bitwise in line.163

Reply:
It's intended. By mdn, button is 0,1,2,3,4,5 but buttons is 1,2,4,8,16
https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/button
https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/buttons

Reply:
good